lib/ASTGen/Sources/ASTGen/CompilerBuildConfiguration.swift

```diff
@@ -321,6 +334,133 @@ public func freeConfiguredRegions(
   UnsafeMutableBufferPointer(start: regions, count: numRegions).deallocate()
 }
 
+/// Cache used when checking for inactive code that might contain a reference
+/// to specific names.
+private struct InactiveCodeContainsReferenceCache {
+  let syntax: SourceFileSyntax
+  let configuredRegions: ConfiguredRegions
+}
+
+/// Search in inactive/unparsed code to look for evidence that something that
+/// looks unused is actually used in some configurations.
+private enum InactiveCodeChecker {
+  case name(String)
+  case tryOrThrow
+
+  /// Search for the kind of entity in the given syntax node.
+  func search(syntax: SourceFileSyntax, configuredRegions: ConfiguredRegions) -> Bool {
+    // If there are no regions, everything is active. This is the common case.
+    if configuredRegions.isEmpty {
+      return false
+    }
+
+    for token in syntax.tokens(viewMode: .sourceAccurate) {
```

Comment:
Using a `SyntaxVisitor` instead of `tokens(viewMode:)` is currently faster because it has performance optimizations to reuse memory allocations, in case this is relevant here.

Reply:
It's harder to stop the visitor from running when we find something, though, whereas I can easily "break" out of loop that walks over tokens.